### ISSUE TITLE
Failed mount folder for smb type

### ISSUE THREAD

**Issue by stiig** [puphpet/puphpet#1930]
Part of config

```
        synced_folder:
            vflsf_kr60co2s660p:
                source: ./
                target: /var/www
                sync_type: smb
                smb:
                    smb_host: ''
                    smb_username: user
                    smb_password: password
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                        - .git/
                    auto: 'true'
                owner: www-data
                group: www-data
```

vagrant up log

```
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'puphpet/debian75-x64' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Loading metadata for box 'puphpet/debian75-x64'
    default: URL: https://atlas.hashicorp.com/puphpet/debian75-x64
==> default: Adding box 'puphpet/debian75-x64' (v2.0) for provider: virtualbox
    default: Downloading: https://atlas.hashicorp.com/puphpet/boxes/debian75-x64/versions/2.0/providers/virtualbox.box
    default: Progress: 100% (Rate: 1077k/s, Estimated time remaining: --:--:--)
==> default: Successfully added box 'puphpet/debian75-x64' (v2.0) for 'virtualbox'!
==> default: Importing base box 'puphpet/debian75-x64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'puphpet/debian75-x64' is up to date...
==> default: Setting the name of the VM: localdev_default_1443694712079_51045
==> default: Preparing SMB shared folders...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 => 5941 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
GuestAdditions versions on your host (5.0.2) and guest (4.3.20) do not match.
stdin: is not a tty
Reading package lists...
Building dependency tree...
Reading state information...
linux-headers-3.2.0-4-amd64 is already the newest version.
The following NEW packages will be installed:
  dkms
0 upgraded, 1 newly installed, 0 to remove and 3 not upgraded.
Need to get 77.4 kB of archives.
After this operation, 196 kB of additional disk space will be used.
Get:1 http://http.debian.net/debian/ wheezy/main dkms all 2.2.0.3-1.2 [77.4 kB]
dpkg-preconfigure: unable to re-open stdin: No such file or directory
Fetched 77.4 kB in 1s (60.4 kB/s)
Selecting previously unselected package dkms.
(Reading database ... 42854 files and directories currently installed.)
Unpacking dkms (from .../dkms_2.2.0.3-1.2_all.deb) ...
Processing triggers for man-db ...
Setting up dkms (2.2.0.3-1.2) ...
Copy iso file C:\Program Files\Oracle\VirtualBox\VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
stdin: is not a tty
mount: block device /tmp/VBoxGuestAdditions.iso is write-protected, mounting read-only
Installing Virtualbox Guest Additions 5.0.2 - guest version is 4.3.20
stdin: is not a tty
Verifying archive integrity... All good.
Uncompressing VirtualBox 5.0.2 Guest Additions for Linux............
VirtualBox Guest Additions installer
Removing installed version 4.3.20 of VirtualBox Guest Additions...
Copying additional installer modules ...
Installing additional modules ...
Removing existing VirtualBox DKMS kernel modules ...done.
Removing existing VirtualBox non-DKMS kernel modules ...done.
Building the VirtualBox Guest Additions kernel modules ...done.
Doing non-kernel setup of the Guest Additions ...done.
You should restart your guest to make sure the new modules are actually used

Installing the Window System drivers
Could not find the X.Org or XFree86 Window System, skipping.
An error occurred during installation of VirtualBox Guest Additions 5.0.2. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
stdin: is not a tty
Restarting VM to apply changes...
==> default: Attempting graceful shutdown of VM...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
==> default: Mounting SMB shared folders...
    default: C:/vagrants/localdev => /var/www
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t cifs -o uid=`id -u www-data`,gid=`getent group www-data | cut -d: -f3`,sec=ntlm,username=PASSWORDHIDDEN,password=PASSWORDHIDDEN ///88f0e9729214559f66d624c8a568c2ed /var/www
mount -t cifs -o uid=`id -u www-data`,gid=`id -g www-data`,sec=ntlm,username=PASSWORDHIDDEN,password=PASSWORDHIDDEN ///88f0e9729214559f66d624c8a568c2ed /var/www

The error output from the last command was:

stdin: is not a tty
mount: wrong fs type, bad option, bad superblock on ///88f0e9729214559f66d624c8a568c2ed,
       missing codepage or helper program, or other error
       (for several filesystems (e.g. nfs, cifs) you might
       need a /sbin/mount.<type> helper program)
       In some cases useful info is found in syslog - try
       dmesg | tail  or so
```

Error:

```
    default: C:/vagrants/localdev => /var/www
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t cifs -o uid=`id -u www-data`,gid=`getent group www-data | cut -d: -f3`,sec=ntlm,username=PASSWORDHIDDEN,password=PASSWORDHIDDEN ///88f0e9729214559f66d624c8a568c2ed /var/www
mount -t cifs -o uid=`id -u www-data`,gid=`id -g www-data`,sec=ntlm,username=PASSWORDHIDDEN,password=PASSWORDHIDDEN ///88f0e9729214559f66d624c8a568c2ed /var/www

The error output from the last command was:

stdin: is not a tty
mount: wrong fs type, bad option, bad superblock on ///88f0e9729214559f66d624c8a568c2ed,
       missing codepage or helper program, or other error
       (for several filesystems (e.g. nfs, cifs) you might
       need a /sbin/mount.<type> helper program)
       In some cases useful info is found in syslog - try
       dmesg | tail  or so
```

**Comment by killua99** [comment]
I fix this error with:

vagrant ssh

sudo apt-get install -y cifs-utils

exit

vagrant reload --provision

It need the cifs-utils in order to get it done. The puphet machines need this library already there.

**Comment by KptnKMan** [comment]
Thanks @killua99 this also worked for me.

**Comment by jtreminio** [comment]
Will be fixed by #1837
